### PR TITLE
[Codespaces] Set --no-localhost by default when logging in

### PR DESCRIPTION
**Pull request by lostintangent**
# Description

In order to make it simpler to use the Firebase CLI within [Codespaces](https://github.com/features/codespaces), this PR simply updates the `firebase login` command to set `--no-localhost` by default, and prevent potential [user confusion](https://twitter.com/shellen/status/1298824809061601280). This change is likely beneficial in other cloud contexts, and so I implemented a new `isCloudEnvironment` util that currently checks for a Codespaces-based environment, but could be easily augmented for others scenarios as well 👍 

### Scenarios Tested

1. ✅ Ran `firebase login` outside of Codespaces, and correctly received the auth callback flow _(Existing behavior)_
1. ✅ Ran `firebase login --no-localhost` outside of Codespaces, and correctly received the auth code flow _(Existing behavior)_
1. ✅ Ran `firebase login` inside of Codespaces, and correctly received the auth callback flow _(New behavior)_
1. ✅ Ran `firebase login --no-localhost` inside of Codespaces, and correctly received the auth callback flow _(Existing behavior)_

### Sample Commands

This change simply updates `firebase login` to set `--no-localhost` by default, when run in the context of a Codespaces.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -447,3 +448,10 @@ export function createDestroyer(server: http.Server): () => Promise<void> {
     return destroyPromise;
   };
 }
+
+/**
+ * Indicates whether the end-user is running the CLI from a cloud-based environment.
+ */
+export function isCloudEnvironment() {
+  return !!env.CODESPACES;
```
Comment (mbleigh) [inline src/utils.ts:456]: nit: for consistency with use elsewhere in the codebase, let's just do `process.env.CODESPACES`

**Comment by lostintangent** [comment]
@googlebot I signed it!

File: src/utils.ts (hunk 1 above)
Comment (lostintangent) [reply src/utils.ts]: Done! 👍

**Comment by lostintangent** [comment]
@mbleigh Thanks for the review! I just updated the changelog and signed the CLA. Let me know if there's anything else I can help with 👍

**Review by bkendall — APPROVED** [review]

**Pull request merged**